lib/AST/InlinableText.cpp

```diff
@@ -189,41 +189,78 @@ struct ExtractInactiveRanges : public ASTWalker {
 };
 } // end anonymous namespace
 
+static void appendRange(
+  SourceManager &sourceMgr, SourceLoc start, SourceLoc end,
+  SmallVectorImpl<char> &scratch) {
+  unsigned bufferID = sourceMgr.findBufferContainingLoc(start);
+  unsigned offset = sourceMgr.getLocOffsetInBuffer(start, bufferID);
+  unsigned endOffset = sourceMgr.getLocOffsetInBuffer(end, bufferID);
+
+  // Strip comments from the chunk before adding it by re-lexing the range.
+  LangOptions FakeLangOpts;
+  Lexer lexer(FakeLangOpts, sourceMgr, bufferID, nullptr, LexerMode::Swift,
+    HashbangMode::Disallowed, CommentRetentionMode::ReturnAsTokens,
+    offset, endOffset);
+
+  SmallVector<SourceRange, 8> nonCommentRanges;
+  SourceLoc nonCommentStart = start;
+  Token token;
+
+  // Re-lex the range, and skip the full text of `tok::comment` tokens.
+  while (true) {
+    lexer.lex(token);
+
+    if (token.is(tok::eof))
+      break;
+
+    if (token.is(tok::comment)) {
+      // Grab the range from the last non-comment token to the beginning of this comment
+      // token.
+      SourceLoc commentLoc = token.getLoc();
+
+      SourceRange range { nonCommentStart, commentLoc };
+      nonCommentRanges.append(1, range);
+
+      // Set the start of the next non-comment range to the end of this token.
+      nonCommentStart = Lexer::getLocForEndOfToken(sourceMgr, token.getLoc());
+    }
+  }
+
+  if (nonCommentStart.isValid() && nonCommentStart != end) {
+    SourceRange range { nonCommentStart, end };
+    nonCommentRanges.append(1, range);
+  }
+
+  for (SourceRange &range : nonCommentRanges) {
+    auto charRange = CharSourceRange(sourceMgr, range.Start, range.End);
+    StringRef text = sourceMgr.extractText(charRange);
+    scratch.append(text.begin(), text.end());
+  }
```

Comment:
Why do you gather all the comment ranges first and then append them to the scratch space instead of appending to the scratch space in the `while` loop above? Is it more performant?

Reply:
Y'know, this went through several revisions before I landed on what I have now, and it took until this comment for me to realize I had written it going forwards again. Fixed.